src/worker/embedded/packagePullerInstaller.py

```diff
@@ -62,17 +62,27 @@ def deps(dirname):
             continue
     return list(rv)
 
-
 def f(event):
     pkg = event["pkg"]
     alreadyInstalled = event["alreadyInstalled"]
+    pip_mirror = event.get("pip_mirror", "")
     if not alreadyInstalled:
         try:
-            subprocess.check_output(
-                ['pip3', 'install', '--no-deps', pkg, '--cache-dir', '/tmp/.cache', '-t', '/host/files'])
+            if pip_mirror == "":
+                subprocess.check_output(
+                    ['pip3', 'install', '--no-deps', pkg, '--cache-dir', '/tmp/.cache', '-t', '/host/files'])
+            else:
+                pip_mirror = pip_mirror.rstrip('/') + '/simple/' # make sure it ends with / and has simple at the end
+                host_start_index = pip_mirror.find('://') + 3
+                host_end_index = pip_mirror.find('/', host_start_index)
+                mirror_host = pip_mirror[host_start_index:host_end_index]
+                cmds = ['pip3', 'install', '--no-deps', pkg, '--cache-dir', '/tmp/.cache', '-t', '/host/files', f"--trusted-host={mirror_host}", f"--index-url={pip_mirror}",  "-vvv"]
```

Comment:
We should avoid copying the original args then adding.  If we need to make future changes, that will have to happen in two places.  Better to have the basic args, then if pip_mirror is set, append the additional ones.